### PR TITLE
Update k8s config

### PR DESCRIPTION
**Pull request by flands**
Also add fully commented config for the field

### CODE REVIEW

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+  #  - grpc (default endpoint = 0.0.0.0:9411)
+  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/zipkinreceiver
+  zipkin:
+
+  #############################################################################
+  # Metrics
+  #############################################################################
+
+  # Enables the k8s cluster receiver with default settings
+  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/k8sclusterreceiver
```
Comment (mstumpfx) [inline cmd/otelcol/config/collector/full_config_linux.yaml:89]:
```suggestion
  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sclusterreceiver
```

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+  # NOTE: It is very likely additional configuration is required
+  k8s_cluster:
+
+  # Enables the kubeletstats receiver with default settings
+  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/kubeletstatsreceiver
```
Comment (mstumpfx) [inline cmd/otelcol/config/collector/full_config_linux.yaml:94]:
```suggestion
  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver
```

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+      value: "zone-1"
+      action: upsert
+    - key: k8s.cluster.name
+      from_attribute: k8s-cluster
+      action: insert
+    - key: redundant-attribute
+      action: delete
+
+  # Enables the resource detection processor with default settings
+  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/resourcedetectionprocessor
```
Comment (mstumpfx) [inline cmd/otelcol/config/collector/full_config_linux.yaml:190]:
```suggestion
  # Full configuration here: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor
```

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+# NOTE: Be sure you understand how processors work
+###############################################################################
+service:
+
+  # Which extensions you want to enable
+  extensions: [health_check, http_forwarder, zpages]
+
+  # Pipelines are data source specific today
+  # Every data source is made up of at least one receiver and one exporter
+  # If processors are defined then order specific is order applied
```
Comment (mstumpfx) [inline cmd/otelcol/config/collector/full_config_linux.yaml:517]:
```suggestion
  # If processors are defined then order specific is the order applied
```

**Review by jrcamp — APPROVED** [review]

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml [hunk 5]
```diff
@@ -180,9 +178,9 @@ spec:
           - "--config=/conf/otel-collector-config.yaml"
           # TODO: Ensure this is set properly
 #           Memory Ballast size should be max 1/3 to 1/2 of memory.
-          - "--mem-ballast-size-mib=683"
+          - "--mem-ballast-size-mib=650"
     # TODO: Ensure this is set properly
-    image: quay.io/signalfx/splunk-otel-collector:0.20.0
+    image: quay.io/signalfx/splunk-otel-collector:latest
```
Comment (mstumpfx) [inline examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:183]: I understand _why_ we would use latest here but I do worry that some will take :latest and use it in their prod deployments, which ain't great for stability.

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+# NOTE: Be sure you understand how processors work
+###############################################################################
+service:
+
+  # Which extensions you want to enable
+  extensions: [health_check, http_forwarder, zpages]
+
+  # Pipelines are data source specific today
+  # Every data source is made up of at least one receiver and one exporter
+  # If processors are defined then order specific is the order applied
```
Comment (flands) [inline cmd/otelcol/config/collector/full_config_linux.yaml:517]:
```suggestion
  # If processors are defined then order specified is the order applied
```

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml (hunk 5 above)
Comment (flands) [reply examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:183]: yeah, good call -- I will add a comment about this

File: examples/kubernetes-yaml/splunk-otel-collector-agent.yaml [hunk 7]
```diff
@@ -122,84 +120,56 @@ data:
...
       containers:
       - command:
           - "/otelcol"
           - "--config=/conf/otel-collector-config.yaml"
           # TODO: Ensure this is set properly
 #           Memory Ballast size should be max 1/3 to 1/2 of memory.
-          - "--mem-ballast-size-mib=683"
+          - "--mem-ballast-size-mib=150"
     # TODO: Ensure this is set properly
     image: quay.io/signalfx/splunk-otel-collector:latest
```
Comment (flands) [inline examples/kubernetes-yaml/splunk-otel-collector-agent.yaml:151]:
```suggestion
    # WARN: It is not recommended to use the latest tag in production deployments!
    image: quay.io/signalfx/splunk-otel-collector:latest
```

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml (hunk 5 above)
Comment (flands) [inline examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:183]:
```suggestion
    # WARN: It is not recommended to use the latest tag in production deployments!
    image: quay.io/signalfx/splunk-otel-collector:latest
```

**Review by mstumpfx — APPROVED** [review]

**Pull request merged**